system/API/ResponseTrait.php

```diff
@@ -328,44 +328,31 @@ protected function format($data = null)
 		// If the data is a string, there's not much we can do to it...
 		if (is_string($data))
 		{
-			$this->setContentType('text/html');
+			// The content type should be text/... and not application/...
+			$contentType = $this->response->getHeaderLine('Content-Type');
+			$contentType = str_replace('application/json', 'text/html', $contentType);
+			$contentType = str_replace('application/', 'text/', $contentType);
```

Comment:
application/* implies machine-processing. All the code I saw suggested that a string value being returned would be most appropriately treated as text/*, so that no Formatter would try to do something to it.

The Format class defines the allowed response formats and implied treatment for arrays, and 'text/html' seems to be the preferred/implied format for strings, *but* XML could be returned as a fully-formed string.
There is a fine line here - what do we automate, what do we assume, and what do we insist the developer take care of. I have triued to take a middle-of-the-road approach.
What would you like to see changed?